docs/adr/0002-database-corruption.md

```diff
@@ -0,0 +1,106 @@
+# Handling Database Corruption
+
+* Status: accepted
+* Date: 2021-06-08
+
+## Context and Problem Statement
+
+Some of our users have corrupt SQLite databases and this makes the related
+component unusable.  The best way to deal with corrupt databases is to simply
+delete the database and start fresh (#2628).  However, we only want to do this
+for persistent errors, not transient errors like programming logic errors, disk
+full, etc.  This ADR deals with 2 related questions:
+
+  * A) When and how do we identify corrupted databases?
+  * B) What do we do when we identify corrupted databases?
+
+## Decision Drivers
+
+* Deleting valid user data should be avoided at almost any cost
+* Keeping a corrupted database around is almost as bad.  It currently prevents
+  the component from working at all.
+* We don't currently have a good way to distinguish between persistent and
+  transient errors, but this can be improved by reviewing telemetry and sentry
+  data.
+
+## Considered Options
+
+* A) When and how do we identify corrupted databases?
+  * 1: Assume all errors when opening a database are from corrupt databases
+  * 2: Check for errors when opening a database and compare against known corruption error types
+  * 3: Check for errors for all database operations and compare against known corruption error types
+* B) What do we do when we identify corrupted databases?
+  * 1: Delete the database file and recreate the database
+  * 2: Move the database file and recreate the database
+  * 3: Have the component fail
+
+## Decision Outcome
+
+* A2: Check for errors when opening a database and compare against known corruption error types
+* B1: Delete the database file and recreate the database
+
+Decision B follows from the choice of A.  Since we're being conservative in
+identifying errors, we can delete the database file with relative confidence.
+
+"Check for errors for all database operations and compare against known
+corruption error types" also seems like a reasonable solution that we may
+pursue in the future, but we decided to wait for now.  Checking for errors
+during opening time is the simpler solution to implement and should fix the
+issue in many cases.  The plan is to implement that first, then monitor
+sentry/telemetry to decide what to do next.
+
+# Pros and Cons of the Options
+
+### A1: Assume all errors when opening a database are from corrupt databases
+* Good, because the sentry data indicates that many errors happen during opening time
+* Good, because migrations are especially likely to trigger corruption errors
+* Good, because it's a natural time to delete the database -- the consumer code hasn't run any queries yet.
+* Bad, because creates the possibility of deleting valid user data.
```

Comment:
I'd even upgrade this to "likely to delete valid user data" - eg, I could imagine OOM etc causing this too.